src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -378,6 +380,9 @@ def update_ag_http_listener(cmd, instance, parent, item_name, frontend_ip=None,
         if firewall_policy is not None:
             instance.firewall_policy = SubResource(id=firewall_policy)
 
+    if host_names is not None:
+        instance.host_names = host_names or None
```

Comment:
under the condition of `host_names` is not None, why need `or` in assignment?

Reply:
"" this one?

Reply:
I mean the default vaule of host_names is None, why test under the if statement ?